src/sidebar/util/test/wrap-react-component-test.js

```diff
@@ -0,0 +1,238 @@
+    assert.called(onClick);
+  });
+
+  it('gets values for injected properties from Angular', () => {
+    const { element } = renderButton();
+    const btnEl = element[0].querySelector('button');
+    assert.equal(btnEl.className, 'btn--dark');
+  });
+
+  it('updates the React component when the Angular component is updated', () => {
+    const { element } = renderButton();
+    const btnEl = element[0].querySelector('button');
+    assert.equal(btnEl.textContent, 'Edit');
+
+    // Change the inputs and re-render.
+    element.scope.label = 'Saving...';
+    element.scope.$digest();
+
+    // Check that the text _of the original DOM element_, was updated.
+    assert.equal(btnEl.textContent, 'Saving...');
+  });
+
+  it('removes the React component when the Angular component is destroyed', () => {
+    // Create parent Angular component which renders a React child.
+    const parentComponent = {
+      controllerAs: 'vm',
+      bindings: {
+        showChild: '<',
+      },
+      template: '<child ng-if="vm.showChild"></child>',
+    };
+
+    // Create a React child which needs to do some cleanup when destroyed.
+    const childUnmounted = sinon.stub();
+    class ChildComponent extends Component {
+      componentWillUnmount() {
+        childUnmounted();
+      }
+    }
+    ChildComponent.propTypes = {};
+
+    angular
+      .module('app', [])
+      .component('parent', parentComponent)
+      .component('child', wrapReactComponent(ChildComponent));
+    angular.mock.module('app');
+
+    // Render the component with the child initially visible.
+    const element = createDirective(document, 'parent', { showChild: true });
+
+    // Re-render with the child removed and check that the React component got
+    // destroyed properly.
+    element.scope.showChild = false;
+    element.scope.$digest();
+
+    assert.called(childUnmounted);
+  });
+
+  it('throws an error if the developer forgets to set propTypes', () => {
+    function TestComponent() {
+      return <div>Hello world</div>;
+    }
+    assert.throws(
+      () => wrapReactComponent(TestComponent),
+      'React component TestComponent does not specify its inputs using "propTypes"'
+    );
+  });
+
+  // Input property checking is handled by React when debug checks are enabled.
+  // This test just makes sure that these checks are working as expected.
+  it('throws an error if property types do not match when component is rendered', () => {
+    const { element } = renderButton();
+    const btnEl = element[0].querySelector('button');
+    assert.equal(btnEl.textContent, 'Edit');
+
+    // Incorrectly set label to a number, instead of a string.
+    element.scope.label = 123;
+    const consoleError = sinon.stub(console, 'error');
+    element.scope.$digest();
+
+    assert.calledWithMatch(
+      consoleError,
+      /Invalid Button `label` of type `number`/
+    );
+  });
+
+  it('throws an error if a callback is passed a non-object argument', () => {
```

Comment:
What exactly is this testing? That the propTypes checking works as expected?

Reply:
Proptypes can only check that a callback is a function. This additionally checks that the callback is passed an object when called. The keys of the object then become variables which can be used in the Angular template of the parent component.